StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+DeferredAPI.swift

```diff
@@ -12,98 +12,105 @@ import Foundation
 @available(iOSApplicationExtension, unavailable)
 @available(macCatalystApplicationExtension, unavailable)
 extension PaymentSheet {
-    static func createPaymentMethodIfNeeded(apiClient: STPAPIClient,
-                                            paymentMethod: STPPaymentMethod?,
-                                            paymentMethodParams: STPPaymentMethodParams?) async throws -> STPPaymentMethod {
-        if let paymentMethod = paymentMethod {
-            return paymentMethod
-        }
-
-        guard let paymentMethodParams = paymentMethodParams else {
-            throw PaymentSheetError.unknown(debugDescription: "paymentMethodParams unexpectedly nil")
-        }
-        return try await apiClient.createPaymentMethod(with: paymentMethodParams)
-    }
-
-    static func handleDeferredIntentConfirmation(deferredIntentContext: DeferredIntentContext,
-                                                 paymentMethod: STPPaymentMethod?,
-                                                 paymentMethodParams: STPPaymentMethodParams?,
-                                                 shouldSavePaymentMethod: Bool) {
+    static func handleDeferredIntentConfirmation(
+        confirmType: ConfirmPaymentMethodType,
+        configuration: PaymentSheet.Configuration,
+        intentConfig: PaymentSheet.IntentConfiguration,
+        authenticationContext: STPAuthenticationContext,
+        paymentHandler: STPPaymentHandler,
+        completion: @escaping (PaymentSheetResult) -> Void
+    ) {
         // Hack: Add deferred to analytics product usage as a hack to get it into the payment_user_agent string in the request to create a PaymentMethod
         STPAnalyticsClient.sharedClient.addClass(toProductUsageIfNecessary: IntentConfiguration.self)
-        Task {
+        Task { @MainActor in
```

Comment:
Was this not running on the main thread?

Reply:
No, not necessarily.  I don't fully understand what the behavior is - Tasks run on the current actor by default, but I'm not sure what that means if you're not explicitly using `@MainActor` or other actors.  In practice, I observed this Task being called on various non-main queues.  Adding `@MainActor` here ensures the closure executes on the main thread.